### PR TITLE
Erms 2136

### PR DESCRIPTION
**Pull request by klodav**

### CODE REVIEW

File: app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy [hunk 1]
```diff
@@ -198,7 +198,7 @@ class PendingChange {
             parsedParams[2] = rdv2.getI10n('value')
         }
         else if (type == 'date') {
-            //def sdf = new java.text.SimpleDateFormat(messageSource.getMessage('default.date.format', null, locale))
+            //java.text.SimpleDateFormat sdf = new java.text.SimpleDateFormat(messageSource.getMessage('default.date.format', null, locale))
```
Comment (croqueGrec09) [inline app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy]: is default.date.format not specified in messages.properties?

File: app/src/groovy/de/laser/helper/DateUtil.groovy [hunk 2]
```diff
@@ -1,26 +1,80 @@
...
+
+    static SimpleDateFormat getSDF_OnlyTime(){
+        getSimpleDateFormatByToken(DATE_FORMAT_ONLYTIME)
+    }
+
+    static Date toDate_NoTime(String value) {
+        (Date) getSDF_NoTime()?.parseObject(value)
+    }
+
+    // --
```
Comment (croqueGrec09) [inline app/src/groovy/de/laser/helper/DateUtil.groovy]: ex escapeService.groovy, change acknowledged

File: app/grails-app/controllers/com/k_int/kbplus/PackageController.groovy [hunk 3]
```diff
@@ -411,7 +413,8 @@ class PackageController extends AbstractDebugController {
             }
 
         } else {
-            def currentDate = new java.text.SimpleDateFormat(message(code: 'default.date.format.notime', default: 'yyyy-MM-dd')).format(new Date())
+            SimpleDateFormat sdf = DateUtil.getSDF_NoTime()
+            def currentDate = sdf?.format(new Date())
```
[the remark this answers is not in the document]
Comment (klodav) [reply app/grails-app/controllers/com/k_int/kbplus/PackageController.groovy]: formatted String != Date-Object

File: app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy [hunk 4]
```diff
@@ -577,7 +577,7 @@ class SubscriptionController extends AbstractDebugController {
                 }
             }
         } else {
-            def currentDate = new java.text.SimpleDateFormat(message(code: 'default.date.format.notime', default: 'yyyy-MM-dd')).format(new Date())
+            def currentDate = (DateUtil.getSDF_NoTime()).format(new Date())
```
[the remark this answers is not in the document]
Comment (klodav) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy]: formatted String != Date-Object

[the remark this answers is not in the document]
Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy]: But why new Date() as argument for format?

File: app/grails-app/controllers/com/k_int/kbplus/PackageController.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/PackageController.groovy]: new Date() as argument for format is a bit weird ...

File: app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy [hunk 5]
```diff
@@ -1327,7 +1327,7 @@ class SubscriptionController extends AbstractDebugController {
 
         result.filterSet = params.filterSet ? true : false
 
-        def sdf = new SimpleDateFormat('yyyy-MM-dd')
+        SimpleDateFormat sdf = new SimpleDateFormat('yyyy-MM-dd')
```
[the remark this answers is not in the document]
Comment (klodav) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy]: Constants currently only exist as a reference for the resolution of language-related translations

**Pull request merged**